commit 249c486ce8e80a9e94d51b4bbf3ccf5d0af57e5e
Author: Siddhesh Poyarekar <siddhesh@sourceware.org>
Date:   Mon Mar 15 17:23:30 2021 +0530

    support: Pass environ to child process
    
    Pass environ to posix_spawn so that the child process can inherit
    environment of the test.
    
    (cherry picked from commit e958490f8c74e660bd93c128b3bea746e268f3f6)

diff --git a/support/support_subprocess.c b/support/support_subprocess.c
index 838eda96ffd6eb9a..2acfc57b7e70732d 100644
--- a/support/support_subprocess.c
+++ b/support/support_subprocess.c
@@ -84,7 +84,7 @@ support_subprogram (const char *file, char *const argv[])
   xposix_spawn_file_actions_addclose (&fa, result.stdout_pipe[1]);
   xposix_spawn_file_actions_addclose (&fa, result.stderr_pipe[1]);
 
-  result.pid = xposix_spawn (file, &fa, NULL, argv, NULL);
+  result.pid = xposix_spawn (file, &fa, NULL, argv, environ);
 
   xclose (result.stdout_pipe[1]);
   xclose (result.stderr_pipe[1]);
